gradoop-flink/src/main/java/org/gradoop/flink/io/impl/csv/indexed/IndexedCSVDataSource.java

```diff
@@ -92,7 +93,10 @@ public LogicalGraph getLogicalGraph() throws IOException {
 
   @Override
   public GraphCollection getGraphCollection() throws IOException {
-    MetaData metaData = MetaData.fromFile(getMetaDataPath(), hdfsConfig);
+    CSVMetaDataSource factory = new CSVMetaDataSource();
+    CSVMetaData metaData = factory.readLocal(getMetaDataPath(), hdfsConfig);
```

Comment:
The method readDistributed would return a DataSet of tuples. To use these tuples the way the method requires them, they would have to be collected, which we should avoid.

Also, the original implementation was identical to readLocal.

Reply:
Oh i see!

Reply:
It might be possible to remove the distributed version, since metadata is written with parallelism 1 anyway and isn't needed as `DataSet` anywhere. Common read/write methods implemented with Flink `FileSystem` would then be able to read/write from both distributed and local file systems without any `DataSet` wrapping.

Alternatively, using a broadcast instead of collecting to unwrap the metadata from the DataSet is also possible and prevents the code duplication of the current local and distributed versions.

Reply:
This discussion can be part of a new issue since this would provide a more general approach. I prefer to place it there. I want this refactoring be part of the next incoming minor release. Therefor i approve the change. However i agree to @timo95. The proposal is correct since the distributed Tuple version is not needed if we just broadcast a dataset with one metadata object.